lib/git-issue/import-export.sh

```diff
@@ -878,16 +878,56 @@ t again
 ' "$1"-header
 }
 
+get_url()
+{
+  local remote=$1
+
+  if [ -z "${remote}" ]
+  then
+    remote="origin"
+  fi
+
+  local url provider user repo
+  url=$(git remote get-url ${remote} 2>&1)
+
+  if [ "${url}" = "fatal: No such remote '${remote}'" ]
+  then
+    printf "%s\n\n" "${url}"
+    usage_import
+  fi
+  echo "${url}"
+}
+
+get_provider()
+{
+  provider=$(echo "$1" | grep -oP "(?<=@).*?(?=\.)")
```

Comment:
This function and the ones' below can simply be `echo | grep` (No need for assignment and second echo).  Even better, use *expr match* or *sed*, rather than *grep* whose use for this purpose isn't common.

Reply:
`This function and the ones' below can simply be echo | grep (No need for assignment and second echo)`

> This function and the ones' below can simply be `echo | grep`
are you suggesting `$(echo | grep -oP "(?<=@).*?(?=\.)")`? If so that doesn't work I am afraid

Reply:
To clarify:
```
get_provider()
{
  echo "$1" | grep -oP "(?<=@).*?(?=\.)"
}
```
or better
```
get_provider()
{
  expr match "$1" 'suitable RE here'
}
```

Reply:
Using `expr match` gives the error https://github.com/koalaman/shellcheck/wiki/SC2003 . Will `sed` or another command like `cut` do?

Reply:
Good point.  It seems that using `:` will not trigger the warning.  Using *sed* is also fine.

Reply:
I made some changes using `sed` and even got it working with `https` URLs. Do you prefer the `sed` styling on line 905, or line 910 and 915? Had trouble with getting the `provider` `sed` working with the extended regex styling